database/database.go

```diff
@@ -192,6 +192,38 @@ func (db *DB) CreateBlockedSkylink(ctx context.Context, skylink *BlockedSkylink)
 	return nil
 }
 
+// CreateBlockedSkylinkBulk creates new blocked skylinks in bulk. It returns the
+// number of created entries.
+func (db *DB) CreateBlockedSkylinkBulk(ctx context.Context, skylinks []BlockedSkylink) (int, error) {
+	// Convenience variables
+	logger := db.staticLogger
+
+	// Ensure all required properties are set on the given blocked skylinks
+	for _, skylink := range skylinks {
+		err := skylink.Validate()
+		if err != nil {
+			return 0, errors.AddContext(err, "unexpected blocked skylink")
+		}
+	}
+
+	// Convert the given array to an interface array
+	docs := make([]interface{}, len(skylinks))
+	for i, doc := range skylinks {
+		docs[i] = doc
+	}
+
+	// Insert all objects in the database
+	opts := options.InsertMany()
+	opts.SetOrdered(false)
```

Comment:
could you explain that this makes it so a single write can fail but that doesn't prevent the other ones from being written?

Reply:
done